pkg/agent/metrics/prometheus.go

```diff
@@ -92,18 +95,21 @@ func InitializePrometheusMetrics(
 		klog.Errorf("Failed to retrieve agent K8S node name: %v", err)
 	}
 
-	gaugeHost := prometheus.NewGauge(prometheus.GaugeOpts{
-		Name:        "antrea_agent_runtime_info",
-		Help:        "Antrea agent runtime info , defined as labels. The value of the gauge is always set to 1.",
-		ConstLabels: prometheus.Labels{"k8s_nodename": nodeName, "k8s_podname": env.GetPodName()},
+	gaugeHost := metrics.NewGauge(&metrics.GaugeOpts{
+		Name:           "antrea_agent_runtime_info",
+		Help:           "Antrea agent runtime info , defined as labels. The value of the gauge is always set to 1.",
+		ConstLabels:    metrics.Labels{"k8s_nodename": nodeName, "k8s_podname": env.GetPodName()},
+		StabilityLevel: metrics.STABLE,
 	})
-	gaugeHost.Set(1)
-	if err := prometheus.Register(gaugeHost); err != nil {
+	if err := legacyregistry.Register(gaugeHost); err != nil {
 		klog.Error("Failed to register antrea_agent_runtime_info with Prometheus")
 	}
+	// This must be after registering the metrics.Gauge as it is lazily instantiated
+	// and will not measure anything unless the collector is first registered.
+	gaugeHost.Set(1)
 
 	ovsStats := newOVSStatManager(ovsBridge, ofClient)
-	if err := prometheus.Register(ovsStats); err != nil {
+	if err := legacyregistry.RawRegister(ovsStats); err != nil {
```

Comment:
Curious why RawRegister is used instead of Register in couple of places?
RawRegister looks to be deprecated from v0.18 k8s release.
https://github.com/kubernetes/kubernetes/commit/214228f0248a19ffa413837035a868410d1b9468